contracts/wrapperContracts/WrapConversionRate.sol

```diff
@@ -2,246 +2,89 @@ pragma solidity 0.4.18;
-        require(tokenControlInfoData.perBlockImbalance.length > index);
-        require(tokenControlInfoData.maxTotalImbalance.length > index);
-        address[] memory signatures;
-        (signatures, nonce) = getDataTrackingParameters(TOKEN_INFO_DATA_INDEX);
-
-        return(
-            tokenControlInfoData.tokens[index],
-            tokenControlInfoData.perBlockImbalance[index],
-            tokenControlInfoData.maxTotalImbalance[index],
-            nonce
-        );
-    }
-
-    function getTokenInfoNumToknes() public view returns(uint numSetTokens) {
-        return tokenControlInfoData.tokens.length;
-    }
-
-    function getTokenInfoData() public view
-        returns(uint nonce, uint numSetTokens, ERC20[] tokenAddress, uint[] maxPerBlock, uint[] maxTotal)
-    {
-        address[] memory signatures;
-        (signatures, nonce) = getDataTrackingParameters(TOKEN_INFO_DATA_INDEX);
-
-        return(
-            nonce,
-            tokenControlInfoData.tokens.length,
-            tokenControlInfoData.tokens,
-            tokenControlInfoData.perBlockImbalance,
-            tokenControlInfoData.maxTotalImbalance);
-    }
-
-    function getTokenInfoSignatures() public view returns (address[] memory signatures) {
-        uint nonce;
-        (signatures, nonce) = getDataTrackingParameters(TOKEN_INFO_DATA_INDEX);
-        return(signatures);
-    }
-
-    function getTokenInfoNonce() public view returns(uint nonce) {
-        address[] memory signatures;
-        (signatures, nonce) = getDataTrackingParameters(TOKEN_INFO_DATA_INDEX);
-        return nonce;
-    }
-
-    //valid duration blocks
-    ///////////////////////
-    function setValidDurationData(uint validDurationBlocks) public onlyOperator {
-        require(validDurationBlocks > 5);
-
-        //update data tracking
-        setNewData(VALID_DURATION_DATA_INDEX);
-
-        validDurationData.durationInBlocks = validDurationBlocks;
-    }
-
-    function approveValidDurationData(uint nonce) public onlyOperator {
-        if (addSignature(VALID_DURATION_DATA_INDEX, nonce, msg.sender)) {
-            // can perform operation.
-            conversionRates.setValidRateDurationInBlocks(validDurationData.durationInBlocks);
-        }
-    }
-
-    function getValidDurationBlocksData() public view returns(uint validDuration, uint nonce) {
-        address[] memory signatures;
-        (signatures, nonce) = getDataTrackingParameters(VALID_DURATION_DATA_INDEX);
-        return(nonce, validDurationData.durationInBlocks);
-    }
-
-    function getValidDurationSignatures() public view returns (address[] memory signatures) {
-        uint nonce;
-        (signatures, nonce) = getDataTrackingParameters(VALID_DURATION_DATA_INDEX);
-        return(signatures);
-    }
-
-    function performAddToken() internal {
-        conversionRates.addToken(addTokenData.token);
+        conversionRates.addToken(token);
 
         conversionRates.addOperator(this);
```

Comment:
Why can the `addOperator` be called multiple times for one address. Isn't there a require check to prevent duplicates, no? 
`require(!operators[newOperator]); // prevent duplicates.`